feat: revamp error handling and improve Sentry integration

tx-nikola opened the pull request:
# Set up Sentry in order to log messages

### Setup the [Sentry](https://sentry.io/) organization and project
- For the project environment, choose “`Vue`”
- Find and copy the Sentry DSN url in this format: “`https://<first_number>@<second_number>.ingest.de.sentry.io/<id>`” - just a whole DSN url
- Set up the env variables:
  - `SENTRY_DSN` # The DSN url mentioned above
  - `SENTRY_ENV` # The environment so we can filter the issues easily -> “localhost” | “production” | “development”
- Then we can get the logged issues from the app
- We choose the issue in the “`Issues`” tab
  - Here we have the information that Sentry picks up
    - For making it easier to resolve issues, for custom handled errors, we also log some of the data related to the errors, such as: `error`, `parentFunctionName`, `parentFunctionParams`, `accountAddress`, `filePath` - in order to find and understand the issue easily (eg. if the params were wrong, which function threw the error, etc.)
  - This information can be found in the `ReferenceError` section of the Issue on the sentry platform


- If we deem some errors not important, and want to limit the number of raised issues (pricing, clutter, etc.) we can additionally ignore them in the config like this:
```
ignoreErrors: [
  "Network error. Check your internet connection and try again.",
  "Network Error",
  "Invalid network",
  "Incorrect wallet network selected",
  "transfer amount exceeds allowance",
],
```
and those errors wouldn’t be logged on the platform.




This PR fixes https://github.com/matter-labs/dapp-portal/issues/217

popzxc commented:
> Set up before merging

Can't we implement it in an additive form, e.g. sentry is only initialized if corresponding env variables are set?
I'm not sure if every project that uses the portal will set up Sentry (at least, right away).

tx-nikola commented:
@popzxc 
Added the changes to omit initializing Sentry SDK if the DSN is not defined in the .env variables.
Also added a small section in the README.md for Sentry.

JackHamer09 commented on views/transactions/Transfer.vue:
```diff
@@ -447,6 +448,13 @@ const totalComputeAmount = computed(() => {
+      parentFunctionName: "totalComputeAmount",
+      parentFunctionParams: [],
+      accountAddress: "",
+      filePath: "views/transactions/Transfer.vue",
```
Isn't there a way to get file path automatically during build time? Like using `import.meta.url` or something like that?

tx-nikola replied:
I tried 2 approaches:
1. Getting the stack trace inside of the sentry like this:
```
let filePath = "";
if (newError.stack) {
  const stackLines = newError.stack.split("\n");
  const callerLine = stackLines[2];
  const match = callerLine.match(/\(([^)]+)\)/);
  if (match) {
    filePath = match[1];
  }
}
```
which, when built, would log the filePath as this: "http://localhost:3000/_nuxt/entry.VMbnGGKk.js:9:9114" which is not descriptive as this is a built file.

2. Using `import.meta.url` as suggested, by passing it as argument in every function call, like this:
```
sentryCaptureException({
  error: error as Error,
  parentFunctionName: "totalComputeAmount",
  parentFunctionParams: [],
  accountAddress: "",
  filePath: import.meta.url,
});
```
which, when built, would log the filePath as this: "http://localhost:3000/_nuxt/Transfer.vue.CwgrnjNF.js" which kinda shows the name of the file, but still not the full path, and we would need to pass it as an argument in every function call, which is not too different than manually defining the file path in every call - it's not used in the function implementation so it's still repetitive.

In this case, I'd leave it as it is, in my opinion it would be better to spot the issue more easily from sentry, rather than use `import.meta.url` as it's not too much of a hassle to just add a path as a string.

JackHamer09 commented on views/transactions/Transfer.vue:
```diff
@@ -447,6 +448,13 @@ const totalComputeAmount = computed(() => {
+      error: error as Error,
+      parentFunctionName: "totalComputeAmount",
+      parentFunctionParams: [],
+      accountAddress: "",
```
Wouldn't it be better to have a composable which wraps sentryCaptureError and it would automatically apply current account address from the onboard store

tx-nikola replied:
Created a composable `useSentryLogger` and updated the code.
